src/main/java/org/walkmod/javalang/compiler/types/TypesLoaderVisitor.java

```diff
@@ -525,19 +565,19 @@ private void loadInnerClass(String name, boolean imported, Node node, List<Symbo
             String internalName = preffix + "$" + suffix;
 
             try {
-                Class<?> clazz = Class.forName(internalName, false, classLoader);
+                ASMClass asmClass = getASMClass(internalName);
```

Comment:
I feal uneasy about making the dependency from this business class to some a liitle bit experimental technology. 
I suggest to decouple using some interface Clazz implemented by asmclass and some interface ClazzLoader providing them. Decouple from technologies.

Reply:
No sure if it is needed. I do not see several implementations of the same interface.

Reply:
I can imagine them but easy enough to add when needed. Adding them now smells like premature optimization and violating KISS. If we avoid leakage of the technology exposing classes thats fine for me.